components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -2477,59 +2478,81 @@ subroutine ma_resuspend_convproc(                           &
 !-----------------------------------------------------------------------
 
 
    do n = 1, ntot_amode
 
-      do ll = 0, nspec_amode(n)
-         if (ll == 0) then
-            la = numptr_amode(n)
-            lc = numptrcw_amode(n) + pcnst
-         else
-            la = lmassptr_amode(ll,n)
-            lc = lmassptrcw_amode(ll,n) + pcnst
-         end if
+      ! TMR number
+      la = numptr_amode(n)
+      lc = numptrcw_amode(n) + pcnst
+      if ((la > 0) .and. (la <= pcnst_extd) .and. (lc > 0) .and. (lc <= pcnst_extd)) then
+         call tmr_tendency(pcnst_extd, dcondt, dcondt_resusp, la, lc, ktop, kbot_prevap)
+      endif
+
+      ! TMR mass
+      do ll = 1, nspec_amode(n)
+         la = lmassptr_amode(ll,n)
+         lc = lmassptrcw_amode(ll,n) + pcnst
+          if ((la > 0) .and. (la <= pcnst_extd) .and. (lc > 0) .and. (lc <= pcnst_extd)) then
+            call tmr_tendency(pcnst_extd, dcondt, dcondt_resusp, la, lc, ktop, kbot_prevap)
+         endif
+      enddo   ! "ll = -1, nspec_amode(n)"
+   enddo      ! "n = 1, ntot_amode"
+
+   return
+   end subroutine ma_resuspend_convproc
+
+!=========================================================================================
+   subroutine tmr_tendency(pcnst_extd, dcondt, dcondt_resusp, &
+                        la, lc, ktop, kbot_prevap)
+!-----------------------------------------------------------------------
+! calculate tendency of TMR
+!-----------------------------------------------------------------------
+   use ppgrid, only: pver
 
-! apply adjustments to dcondt for pairs of unactivated (la) and 
-! activated (lc) aerosol species
-         if ( (la <= 0) .or. (la > pcnst_extd) ) cycle
-         if ( (lc <= 0) .or. (lc > pcnst_extd) ) cycle
+   implicit none
 
-         do k = ktop, kbot_prevap ! REASTER 08/05/2015
+   ! arguments (note:  TMR = tracer mixing ratio)
+   integer,  intent(in)    :: pcnst_extd ! cloudborne aerosol dimension [unitless]
+   real(r8), intent(inout) :: dcondt(pcnst_extd,pver) ! overall TMR tendency from convection [kg/kg/s]
+   real(r8), intent(inout) :: dcondt_resusp(pcnst_extd,pver) ! portion of TMR tendency due to resuspension [kg/kg/s]
+   integer, intent(in)     :: la, lc            ! indices
+   integer,  intent(in)    :: ktop, kbot_prevap ! indices of top and bottom cloud levels ! REASTER 08/05/2015
+
+   ! local variables
+   integer  :: k
+   real(r8) :: qdota, qdotc, qdotac  ! working variables (MR tendencies)
+
+   do k = ktop, kbot_prevap ! REASTER 08/05/2015
             qdota = dcondt(la,k)
             qdotc = dcondt(lc,k)
             qdotac = qdota + qdotc
-
-! cam5 approach
+            ! cam5 approach
             dcondt(la,k) = qdotac
             dcondt(lc,k) = 0.0
-
             dcondt_resusp(la,k) = (dcondt(la,k) - qdota)
             dcondt_resusp(lc,k) = (dcondt(lc,k) - qdotc)
-         end do
-
-      end do   ! "ll = -1, nspec_amode(n)"
-   end do      ! "n = 1, ntot_amode"
+   enddo
```

Comment:
This code probably can be shortened like this:
```
do k = ktop, kbot_prevap ! REASTER 08/05/2015
            dcondt(la,k)              =    dcondt(la,k) + dcondt(lc,k)
            dcondt_resusp(la,k) =    dcondt(lc,k)
            dcondt_resusp(lc,k) = - dcondt(lc,k)

            dcondt(lc,k) = 0.0
enddo
```
I haven't verified it. If you agree, would you please test it to ensure that it works and results are BFB?